evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -0,0 +1,180 @@
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from evalml.pipelines import TimeSeriesRegressionPipeline
+
+
+@pytest.fixture
+def ts_data():
+    X, y = pd.DataFrame({"features": range(101, 132)}), pd.Series(range(1, 32))
+    y.index = pd.date_range("2020-10-01", "2020-10-31")
+    X.index = pd.date_range("2020-10-01", "2020-10-31")
+    return X, y
+
+
+@pytest.mark.parametrize("pipeline_class", [TimeSeriesRegressionPipeline])
+@pytest.mark.parametrize("components", [["One Hot Encoder"],
+                                        ["Delayed Feature Transformer", "One Hot Encoder"]])
+def test_time_series_pipeline_init(pipeline_class, components):
+
+    class Pipeline(pipeline_class):
+        component_graph = components + ["Random Forest Regressor"]
+
+    if "Delayed Feature Transformer" not in components:
+        pl = Pipeline({'pipeline': {"gap": 3, "max_delay": 5}})
+        assert "Delayed Feature Transformer" not in pl.parameters
+        assert pl.parameters['pipeline'] == {"gap": 3, "max_delay": 5}
+    else:
+        parameters = {"Delayed Feature Transformer": {"gap": 3, "max_delay": 5},
+                      "pipeline": {"gap": 3, "max_delay": 5}}
+        pl = Pipeline(parameters)
+        assert pl.parameters['Delayed Feature Transformer'] == {"gap": 3, "max_delay": 5,
+                                                                "delay_features": True, "delay_target": True}
+        assert pl.parameters['pipeline'] == {"gap": 3, "max_delay": 5}
+
+    assert Pipeline(parameters=pl.parameters) == pl
+
+    with pytest.raises(ValueError, match="gap and max_delay parameters cannot be omitted from the parameters dict"):
+        Pipeline({})
+
+
+@pytest.mark.parametrize("only_use_y", [True, False])
+@pytest.mark.parametrize("include_delayed_features", [True, False])
+@pytest.mark.parametrize("gap,max_delay", [(1, 2), (2, 2), (7, 3), (2, 4)])
```

Comment:
Do we have coverage where `gap` is set to 0 ("given historical features and today's features, predict today's target"), or where `max_delay` is set to 0 ("given today's features, predict a future target") or 1?

I think we should add this to the tests in this file, fit/predict/score

Reply:
Added these combinations of gap and max_delay to the tests!

```python
(0, 0), (1, 0), (0, 2), (1, 1)
```